both/methods.js

```diff
@@ -153,16 +153,25 @@ Meteor.methods({
 
     Meteor.call('refreshRoundCodes', competitionId, eventCode);
   },
-  addNonEventRound: function(competitionId, round) {
+  addScheduleEvent: function(competitionId, eventData, roundId) {
     check(competitionId, String);
     throwIfCannotManageCompetition(this.userId, competitionId);
-    return Rounds.insert({
+
+    var round = Rounds.findOne(roundId);
+
+    return ScheduleEvents.insert({
       competitionId: competitionId,
-      title: round.title,
-      startMinutes: round.startMinutes,
-      durationMinutes: round.durationMinutes,
+      roundId: (round ? round._id : null),
+      title: (round ? round.displayTitle() : eventData.title),
```

Comment:
So after sleeping on it, I think a function like this in `ScheduleEvents` is the way to go;

    displayTitle: function() {
      return this.roundId ? Rounds.findOne(this.roundId).displayTitle() : this.title;
    }

Two more thoughts

1. It's not hard to get rid of `refreshRoundCodes`. If the round knows how many rounds there are total, it knows if it's the last one and can compute the roundCode when asked. You do this either by adding a `totalRounds` field, or doing a live count query each time.

2. Reading **Some Event: Second round** in the schedule leaves out important info. **Some Event: Round 2 of 3** would be better.